cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -169,6 +182,93 @@ protected void setMetricsEnabled(boolean isMetricsEnabled) {
+            defaultSettings.put("log4j.logger.kafka", "INFO");
+            defaultSettings.put("log4j.logger.org.apache.kafka", "INFO");
+            defaultSettings.put("log4j.logger.kafka.request.logger", "WARN, CONSOLE");
+            defaultSettings.put("log4j.logger.kafka.network.Processor", "OFF");
+            defaultSettings.put("log4j.logger.kafka.server.KafkaApis", "OFF");
+            defaultSettings.put("log4j.logger.kafka.network.RequestChannel$", "WARN");
+            defaultSettings.put("log4j.logger.kafka.controller", "TRACE");
+            defaultSettings.put("log4j.logger.kafka.log.LogCleaner", "INFO");
+            defaultSettings.put("log4j.logger.state.change.logger", "TRACE");
+            defaultSettings.put("log4j.logger.kafka.authorizer.logger", "WARN");
+        }
+        return defaultSettings;
+    }
+
+    private String createLogConfigMap(Map<String, Object> newSettings) {
+        StringBuilder sb = new StringBuilder();
+        sb.append("# Do not change this generated file. To change logging settings please change cluster configmap.\n");
+        sb.append("# To change kafka log levels, use 'kafka-log-config' field.\n");
+        sb.append("# To change zookeeper log levels, use 'zookeeper-log-config' field.\n");
+        newSettings.forEach((key, value) -> {
+            sb.append(key);
+            sb.append("=");
+            sb.append(value);
+            sb.append("\n");
+        });
+        return sb.toString();
+    }
+
+    protected void setLogConfig(JsonObject logConfig) {
+        Map<String, Object> origin;
+        Map<String, Object> newSettings;
+
+        if (logConfig == null) {
+            // field is not in the cluster CM
+            newSettings = getDefaultLogConfig();
+            this.logConfig = createLogConfigMap(newSettings);
+        } else {
+            origin = logConfig.getMap();
+            if (origin.get("fromConfigMap") != null) {
+                // we want to use custom CM
+                JsonObject json = JsonObject.mapFrom(origin.get("fromConfigMap"));
+                String cmName = json.getString("name");
+                if (cmName == null) {
+                    throw new InvalidConfigMapException("Field fromConfigMap must contain \"name\"");
+                }
+                ConfigMap customCM = new ConfigMapOperator(vertx, new DefaultKubernetesClient()).get(namespace, cmName);
+                if (customCM == null) {
+                    throw new InvalidConfigMapException("Custom config map " + cmName + " not found!");
+                }
+                this.logConfig = customCM.getData().get(LOG_CONFIG_FILE);
+            } else {
+                // update fields otherwise use default values
+                newSettings = getDefaultLogConfig();
+                for (Map.Entry<String, Object> entry : origin.entrySet()) {
+                    newSettings.replace(entry.getKey(), entry.getValue().toString());
```

Comment:
So ... my original idea was that we would offer selected fields which will be validated. And if someone wants to provide a completely free configuration, he would use the config map. However now, you are treating the content here as a "free map" without any validation. That makes the config map import something what is not needed anymore since it doesn't add anything. I think we should decide one way or another.

If we think that including the log configuration directly into the CM / CRD is the preferred approach I'm fine with it ... but then we should probably remove the config map feature.